apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -875,24 +906,20 @@ export async function updateLti13Scores(
         },
       };
 
-      const res = await fetchRetry(assessment.lti13_lineitem_id_url + '/scores', {
-        method: 'POST',
-        headers: {
-          Authorization: `Bearer ${token}`,
-          'Content-type': 'application/vnd.ims.lis.v1.score+json',
-        },
-        body: JSON.stringify(score),
-      });
-
-      if (res.ok) {
+      try {
+        await fetchRetry(assessment.lti13_lineitem_id_url + '/scores', {
+          method: 'POST',
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-type': 'application/vnd.ims.lis.v1.score+json',
+          },
+          body: JSON.stringify(score),
+        });
         counts.success++;
-      } else {
+      } catch (error) {
         counts.error++;
-        const results = await res.json();
-
-        // Default to showing the whole error
-        job.warn(`\t${res.statusText}`);
-        job.warn(`\t${JSON.stringify(results)}`);
+        job.warn(`\t${error.message}`);
+        job.verbose(error.data.body);
```

Comment:
This is highly unsafe. What if this is something other than the `AugmentedError` that you constructed? Then `data` won't exist and your error handling will in fact cause an error. You'll want a conditional here.